electricitymap/contrib/lib/models/events.py

```diff
@@ -108,6 +128,21 @@ def __setattr__(
             value = None
         return super().__setattr__(name, value)
 
+    def _correct_negative_value(
+        self, mode: str, value: Optional[float], correct_negative_with_zero: bool
+    ) -> Union[float, None]:
+        """
+        Corrects a negative value by setting it to None or 0.
+        This method also keeps track of the modes that have been corrected.
+        """
+        if value is not None and value < 0:
+            self._corrected_negative_values.add(mode)
+            if correct_negative_with_zero == True:
```

Comment:
```suggestion
            value = 0 if correct_negative_with_zero else None
```

Reply:
Might be easier one line

Reply:
Sure thing, I thought we might want to add something else here in the future (like record the value) but it would be trivial to change back.